app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -343,21 +345,29 @@ private static PokemonNameCorrector initCorrectorFromPrefs(PokeInfoCalculator po
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        if (intent != null && intent.hasExtra(KEY_TRAINER_LEVEL)) {
-            trainerLevel = intent.getIntExtra(KEY_TRAINER_LEVEL, 1);
-            statusBarHeight = intent.getIntExtra(KEY_STATUS_BAR_HEIGHT, 0);
-            batterySaver = intent.getBooleanExtra(KEY_BATTERY_SAVER, false);
-            makeNotification(this);
-            createInfoLayout();
-            createIVButton();
-            createArcPointer();
-            createArcAdjuster();
-            /* Assumes MainActivity initialized ScreenGrabber before starting this service. */
-            if (!batterySaver) {
-                screen = ScreenGrabber.getInstance();
-                startPeriodicScreenScan();
-            } else {
-                screenShotHelper = ScreenShotHelper.start(Pokefly.this);
+        if (intent != null) {
```

Comment:
@sarav I see now on my PC (from mobile it was harder to read)
I'm going to refactor that code to follow your example above! It will be much cleaner!